libs/sdk-ui-dashboard/src/model/events/insight.ts

```diff
@@ -403,6 +403,47 @@ export const isDashboardInsightWidgetChanged = eventGuard<DashboardInsightWidget
 //
 //
 
+/**
+ * This event is emitted after execution of an insight widget starts.
+ *
+ * @alpha
+ */
+export interface DashboardInsightWidgetExecutionStarted extends IDashboardEvent {
+    readonly type: "GDC.DASH/EVT.INSIGHT_WIDGET.EXECUTION_STARTED";
+    readonly payload: {
+        widgetRef: ObjRef;
+        insightRef: ObjRef;
+    };
```

Comment:
could use comments; + additional things to consider:

-  it would be good if this contained more information about the execution itself. is it feasible to add it? the prepared execution would be useful; is it possible to push that info? or if not the whole prep execution, perhaps the effective filters? the idea is that whoever gets the event can possibly replicate the execution to get the same data
 
-  i think that in order to correctly pair the start/success/fail events, we should have some unique execution identifier here; iirc it is not uncommon that during some ui actions the widget gets re-rendered and execution gets triggered twice in succession. how would the receiver (reliably) know which exec finished once he receives the finished/failed event?